### PR TITLE
refactor: create runtime_defs.h

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

File: src/nvim/runtime_defs.h [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    sctx_T *sctx;       ///< script and modeline info
+    ufunc_T *ufunc;     ///< function info
+    AutoPatCmd *aucmd;  ///< autocommand info
+    except_T *except;   ///< exception info
+  } es_info;
+} estack_T;
+
+/// Stack of execution contexts.  Each entry is an estack_T.
+/// Current context is at ga_len - 1.
+extern garray_T exestack;
```
Comment (zeertzjq) [inline src/nvim/runtime_defs.h:35]: This shouldn't be here.

File: src/nvim/runtime.h [hunk 2]
```diff
@@ -1,120 +1,7 @@
...
-#define DIP_ALL 0x01    // all matches, not just the first one
-#define DIP_DIR 0x02    // find directories instead of files
-#define DIP_ERR 0x04    // give an error message when none found
-#define DIP_START 0x08  // also use "start" directory in 'packpath'
-#define DIP_OPT 0x10    // also use "opt" directory in 'packpath'
-#define DIP_NORTP 0x20  // do not use 'runtimepath'
-#define DIP_NOAFTER 0x40  // skip "after" directories
-#define DIP_AFTER   0x80  // only use "after" directories
-#define DIP_DIRFILE 0x200  // find both files and directories
+#include "nvim/option_defs.h"  // IWYU pragma: export
```
Comment (zeertzjq) [inline src/nvim/runtime.h:3]: Why `export` instead of `keep`?

Comment (dundargoc) [reply src/nvim/runtime.h:3]: Just a mistake.

File: src/nvim/runtime_defs.h (hunk 1 above)
Comment (dundargoc) [reply src/nvim/runtime_defs.h:35]: Yeah, you're right.

**Pull request merged**